docs/spec.md

```diff
@@ -170,6 +173,82 @@ Informs the client that channels are no longer available.
 }
 ```
 
+### Client Advertise
+
+- Informs the server about available client channels.
+
+#### Fields
+
+- `op`: string `"client_advertise"`
```

Comment:
> On the other hand, what do you think about using `"op": "advertise"`? Since the message is sent from client to server, there's no ambiguity. Similarly ops `unadvertise` and `message_data` (or just `message`?) below.

I like your proposal. Maybe we could name the sending message operation `publish` (similar to rosbridge). This would be more consistent with all operations being verbs.

Reply:
> all operations being verbs

not really true for the server->client message op, but since it's a binary op, the opcode is just `1` :)

`publish` is fine.